rtos/Mutex.h

```diff
@@ -76,14 +76,21 @@ class Mutex : private mbed::NonCopyable<Mutex> {
      */
     osStatus unlock();
 
+    /** Check to see if the current thread is the owner of this mutex
+      @return  true if the mutex is owned by the current thread, false otherwise.
+     */
+    bool is_owner();
```

Comment:
I think `am_owner()` would be a better name - can't really use third-person without specifying who the third person is. Yes, the comment does, but nothing in the API name does. But see other comment about whether this would better be a `get_owner`.

Reply:
Added osThreadId Mutex::get_owner() const to be more generic